core/server/src/main/java/alluxio/worker/block/BlockLockManager.java

```diff
@@ -52,7 +52,9 @@ public void close() {}
 
     @Override
     protected ClientRWLock createNewResource() {
-      return new ClientRWLock();
+      ClientRWLock lock = new ClientRWLock();
+      lock.addReference();
```

Comment:
Good call - I was worried that the lock might be returned to the block pool if we didn't immediately add a reference to it, but it's actually safe because locks can't be returned to the lock pool until they have been added to `mLocks`